tests/test_image_vault_utils.cpp

```diff
@@ -83,7 +83,7 @@ TEST_F(TestImageVaultUtils, copy_to_dir_copys_to_dir)
 TEST_F(TestImageVaultUtils, compute_hash_throws_when_cant_read)
 {
     QBuffer buffer{}; // note: buffer is not opened
-    MP_EXPECT_THROW_THAT(MP_IMAGE_VAULT_UTILS.compute_hash(buffer),
+    MP_EXPECT_THROW_THAT(auto _ = MP_IMAGE_VAULT_UTILS.compute_hash(buffer),
```

Comment:
You could use `std::ignore` for this, but really, isn't this a signal that the function should not be `[nodiscard]`? I know recommendations sometimes ignore exceptions (including clang-tidy), but we don't need to :slightly_smiling_face:

IMO, if a function returns and throws, and if it is a reasonable use for the client to just call to check for validity, then it shouldn't be nodiscard.

Reply:
`std::ignore` would be good here. I agree with your general opinion but for this specific case I'm going to disagree. I don't think it's reasonable for a user to call this function without caring about the return. There are much much much better ways to check validity, and I'm not sure why you'd only check validity (other than for the sake of testing).

Reply:
OK, makes sense.